src/components/auth/VerificationBanner.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+
+import { Alert } from '@material-ui/lab';
+import { Collapse } from '@material-ui/core';
+import { KeyboardArrowDown, KeyboardArrowUp } from '@material-ui/icons';
+import { useSnackbar } from 'notistack';
+
+import { LoadingOverlay } from '@app/contexts/ui/LoadingOverlay';
+import { sendAccountVerificationEmail } from '@app/firebase/firebase';
+import { useUser } from '@app/contexts/services/User';
+
+import Link from '../common/Link';
+
+import useStyles from './_jss/VerificationBanner.jss';
+
+const VerificationBanner: React.FC = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const snackbarContext = useSnackbar();
+  const { user } = useUser();
+  const classes = useStyles();
+
+  const toggleCollapse = (): void => {
+    setOpen(prevOpen => !prevOpen);
+  };
+
+  const handleVerificationLink = async (): Promise<void> => {
+    try {
+      setLoading(true);
+      await sendAccountVerificationEmail({ email: user?.email });
+      snackbarContext.enqueueSnackbar('Verification email sent.', { variant: 'success' });
+    } catch {
+      snackbarContext.enqueueSnackbar('Unable to send verification email. Please try again', {
+        variant: 'error',
+      });
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const emailVerificationBanner = (
+    <Alert
+      key="verify-email"
+      severity="warning"
+      icon={false}
+      classes={{ message: classes.verificationBannerMessage }}
+    >
+      Please verify your email by following the link in your inbox.{' '}
+      <Link underline="always" onClick={handleVerificationLink}>
+        Resend verification link
+      </Link>
+    </Alert>
+  );
+
+  const hallVerificationBanner = (
+    <Alert
+      key="verify-hall"
+      severity="warning"
+      icon={false}
+      classes={{ message: classes.verificationBannerMessage }}
+    >
+      Your hall isn't verified. Please ask your hall president to verify you.
```

Comment:
Maybe:
```suggestion
      You have not been verified with your hall. Please ask your hall president to verify you.
```
"Your hall isn't verified" sounds to me like there's an issue with the entire hall

Reply:
520079e7516f8ef5a381b9002ba88afdfd151d03